### DISCUSSION TITLE
Cached object refresh

### DISCUSSION THREAD

**Discussion by jacob7395** [ZiggyCreatures/FusionCache#330] [Q&A] [answered]
Hey,

There might be a way to support this but I am struggling to find it, and I have idea for working around it but it's not my ideal.

I have a cache that is holding object that currently have a 10 min expiry these object are either in a completed or in progress state. Ideally I would leave the timeout at 10 mins (possibly even reduce it) but when the object expires check the state, if it's still in progress re cache/extend the duration of the object in the cache.

I have looked at the docs for plugins and events but and implemented something that looks like it might work but I'm a bit unclear on a few things.

1. Is there a way to support this more natively without needing to re cache the object?
2. Do event's/plugins have async support?
3. What is the sender object in the event, the docs say `the source of the event` but is that going to be the cache? or the object in the cache?
4. When using the subscribing to the Events.Remove will the object have already been removed from the cache?
5. If I try to get the instance from the cache then add it again will this have unexpected consequences?

Thank you in advance to anyone who can advice on this.

**Comment by jacob7395** [comment]
I look at the code and I think I answered some of my questions;

2. No, is this something you would be open too?
3. Sender is the cache that removed the entity
4. Yes, the entity has been removed
5. It wont work the entity has been removed

**Comment by jodydonetti** [comment, marked as answer]
Hi @jacob7395 , that was fast 😅

> Ideally I would leave the timeout at 10 mins (possibly even reduce it) but when the object expires check the state, if it's still in progress re cache/extend the duration of the object in the cache.

if I got this right you may use [Conditional Refresh](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/ConditionalRefresh.md) but without the ETag part: basically you can look at `ctx.StaleValue`, do your own check on the progress, and if you need to cache the same thing for a little bit more you can just return `ctx.NotModified()`.

You can also adapt the `Duration` or other options based on your own logic thanks to [Adaptive Caching](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/AdaptiveCaching.md), if you like.

Of course all the other features of FusionCache like fail-safe, soft timeouts, eager refresh and so on will keep working as usual, so you can mix them together as you like.

Hope this helps, let me know.

**Reply by jacob7395** [reply]
They sound like they will do the trick and give the docs a read tomorrow and update if there are any issues.

Thanks for the quick reply 😊
